rfcs/0114-nixos-code-of-conduct.md

```diff
@@ -0,0 +1,112 @@
+people in socially marginalized groups.
+- Private harassment is also unacceptable. No matter who you are, if you feel
+you have been or are being harassed or made uncomfortable by a community member,
+please contact one of the channel ops or any of the NixOS moderation team immediately.
+Whether you’re a regular contributor or a newcomer, we care about making this community
+a safe place for you and we’ve got your back.
+- Likewise any spamming, trolling, flaming, baiting or other attention-stealing behavior is not welcome.
+
+# Examples and Interactions
+[examples-and-interactions]: #examples-and-interactions
+
+There have been a few spectacular failures of communication and good faith within the community.
+Most notable recent examples include [ad-hominem attacks in a PR review](https://github.com/NixOS/nixpkgs/pull/120729#discussion_r621885348)
+and associated [block evasion discourse thread](https://discourse.nixos.org/t/github-block-evasion-is-not-acceptable/12763),
+heated discussions in [RFC #98](https://github.com/NixOS/rfcs/pull/98) and [RFC #111](https://github.com/NixOS/rfcs/pull/111).
+Each of these incidents have caused significant discussion on IRC (when it was still official),
+matrix, discourse, and would bleed over into github and even non-official platforms.
+These discussions are generally very polarizing, and causes an enormous amount
+of emotional and mental stress to those involved.
+These incidents are also very embarrassing for the greater nix community,
+and not having a way to identify disruptive before it becomes a heated
+issue is detrimental to the health of the community.
+
+In conjunction with a moderation team (out-of-scope for this RFC), these incidents could have been
+better arbitrated as to have a more satisfactory resolution before escalation. This
+Code of Conduct will better equip the moderation process by providing clear expectations
+for behavior within the community.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+- There are potentially more "rules" for the community to follow. However, many of these rules
+are enforced already under the current moderation team, but without an explicit definition.
+
+# Alternatives
+[alternatives]: #alternatives
+
+- #98 Also provides an opinionated values and goals for the moderation team. However,
+these rules are less well-defined, and allow for broader interpretations
+of unacceptable behavior.
+
+- #102 Codifies some of the existing moderation practices, and uses the
+existing [community ethos](https://nixos.org/community/index.html) as moderation criteria.
+
+- Use existing [community ethos](https://nixos.org/community/index.html) as moderation criteria.
+However, this only includes inclusive statements, and doesn't define unacceptable behavior.
+
+# Unresolved questions
+[unresolved]: #unresolved-questions
+
+# Future work
+[future]: #future-work
+
+- Ratification of a "NixOS moderation team" as described in the second to last bullet point.
```

Comment:
This can be removed, since RFC 102 was merged and we have a moderation team.

Reply:
```suggestion
```